server/events/runtime/run_step_runner.go

```diff
@@ -34,6 +34,9 @@ func (r *RunStepRunner) Run(ctx models.ProjectCommandContext, command []string,
 		fmt.Sprintf("ATLANTIS_TERRAFORM_VERSION=%s", tfVersion),
 		fmt.Sprintf("DIR=%s", path),
 		fmt.Sprintf("PLANFILE=%s", filepath.Join(path, GetPlanFilename(ctx.Workspace, ctx.ProjectConfig))),
+		fmt.Sprintf("REPO_OWNER=%s", ctx.BaseRepo.Owner),
+		fmt.Sprintf("REPO_NAME=%s", ctx.BaseRepo.Name),
+		fmt.Sprintf("BRANCH_NAME=%s", ctx.Pull.Branch),
```

Comment:
Let's get everything in there:
```go
	baseEnvVars := os.Environ()
	customEnvVars := map[string]string{
		"WORKSPACE":                  ctx.Workspace,
		"ATLANTIS_TERRAFORM_VERSION": tfVersion,
		"DIR":              path,
		"PLANFILE":         filepath.Join(path, GetPlanFilename(ctx.Workspace, ctx.ProjectConfig)),
		"BASE_REPO_NAME":   ctx.BaseRepo.Name,
		"BASE_REPO_OWNER":  ctx.BaseRepo.Owner,
		"HEAD_REPO_NAME":   ctx.HeadRepo.Name,
		"HEAD_REPO_OWNER":  ctx.HeadRepo.Owner,
		"HEAD_BRANCH_NAME": ctx.Pull.Branch,
		"PULL_NUM":         fmt.Sprintf("%d", ctx.Pull.Num),
		"PULL_AUTHOR":      ctx.Pull.Author,
	}
	finalEnvVars := baseEnvVars
	for key, val := range customEnvVars {
		finalEnvVars = append(finalEnvVars, fmt.Sprintf("%s=%s", key, val))
	}
	cmd.Env = finalEnvVars
```

We also need to update the tests. Can do something like the last case of `run_step_runner_test.go`:
```go
		{
			Command: "echo workspace=$WORKSPACE version=$ATLANTIS_TERRAFORM_VERSION dir=$DIR planfile=$PLANFILE",
			ExpOut:  "workspace=myworkspace version=0.11.0 dir=$DIR planfile=$DIR/myworkspace.tfplan\n",
		},
```

Reply:
Sounds good to me. I'll make the additions and push it up later this evening.

Reply:
Oh and we'll also need to update the docs: https://github.com/runatlantis/atlantis/blob/master/runatlantis.io/docs/atlantis-yaml-reference.md#custom-run-command

Reply:
Got it.